sphinx_qt_documentation/__init__.py

```diff
@@ -9,201 +9,39 @@
-            "pyqtSignal",
-        ),
-    }
-    for module_name, (version, signal_name) in name_mapping.items():
-        try:
-            core = importlib.import_module(f"{module_name}.QtCore")
-            signal = getattr(core, signal_name)
-            return signal, version()
-        except (ModuleNotFoundError, ImportError):
-            continue
-    raise RuntimeError("No Qt bindings found")
-
-
-Signal, QT_VERSION = _get_signal_and_version()
-
-# TODO add response to
-#  https://stackoverflow.com/questions/47102004/how-to-properly-link-to-pyqt5-documentation-using-intersphinx
-
-signal_slot_uri = {
-    "Qt5": "https://doc.qt.io/qt-5/signalsandslots.html",
-    "PySide2": "https://doc.qt.io/qtforpython/overviews/signalsandslots.html",
-    "PyQt5": "https://www.riverbankcomputing.com/static/Docs/PyQt5/signals_slots.html",
-}
-
-signal_name_dict = {"Qt5": "Signal", "PySide2": "Signal", "PyQt5": "pyqtSignal"}
-
-slot_name = {"Qt5": "Slot", "PySide2": "Slot", "PyQt5": "pyqtSlot"}
-
-type_translate_dict = {
-    "class": ["class", "enum"],
-    "meth": ["method", "signal"],
-    "mod": ["module"],
-}
-
-signal_pattern = re.compile(r"((\w+\d?\.QtCore\.)|(QtCore\.)|(\.))?(pyqt)?Signal")
-slot_pattern = re.compile(r"((\w+\d?\.QtCore\.)|(QtCore\.)|(\.))?(pyqt)?Slot")
-
-
-# noinspection PyUnusedLocal
-def missing_reference(
-    app: Sphinx, env: BuildEnvironment, node: Element, contnode: TextElement
-) -> Optional[nodes.reference]:
-    """Linking to Qt documentation."""
-    target: str = node["reftarget"]
-    inventories = InventoryAdapter(env)
-    objtypes: Optional[List[str]] = None
-    if node["reftype"] == "any":
-        # we search anything!
-        objtypes = [
-            f"{domain.name}:{objtype}"
-            for domain in env.domains.values()
-            for objtype in domain.object_types
-        ]
-        domain = None
-    else:
-        domain = node.get("refdomain")
-        if not domain:
-            # only objects in domains are in the inventory
-            return None
-        objtypes = env.get_domain(domain).objtypes_for_role(node["reftype"])
-        if not objtypes:
-            return None
-        objtypes = [f"{domain}:{objtype}" for objtype in objtypes]
-    if target.startswith("PySide2"):
-        head, dot, tail = target.partition(".")
-        target = "PyQt5" + dot + tail
-    if signal_pattern.match(target):
-        uri = signal_slot_uri[app.config.qt_documentation]
-        dispname = signal_name_dict[app.config.qt_documentation]
-        version = QT_VERSION
-    elif slot_pattern.match(target):
-        uri = signal_slot_uri[app.config.qt_documentation]
-        dispname = slot_name[app.config.qt_documentation]
-        version = QT_VERSION
-    else:
-        target_list = [target, "PyQt5." + target]
-        target_list += [
-            name + "." + target
-            for name in inventories.named_inventory["PyQt5"]["sip:module"].keys()
```

Comment:
While the new form maybe shouldn't be hardcoded to "PyQt5", it does seem that we shouldn't be checking for any hit anywhere in the inventory otherwise we end up handling non-Qt items.

Reply:
it is deleted part of code

Reply:
Yeah, I was just referencing a place that I think related to avoiding the issue in the previous implementation.